examples/Main.hs

```diff
@@ -33,10 +35,10 @@ main = putCss logo
          forM_ [0..3] $ \x ->
            forM_ [0..3] $ \y ->
              do let idx = (pack . show) (1 + y * 4 + x)
-                    clr = cycle cs !! fromInteger y
```

Comment:
This is to make `y` typecheck as a `Double`, right? Is it possible to keep `y` an `Integer` and change other types accordingly?

Reply:
No, `y` here typechecks as an `Integer`. The problem here is that `cycle` takes `Int`, not `Integer`... So we need to convert it. Annoying. Overall, I think it wouldn't be a bad idea to just directly specify `[0..3] :: [Integer]`, because it can be confusing what variables like `x`, `y`, `s` and `m` here typecheck to.

Reply:
In fact, `x` and `y` typecheck as `Integer`s if `s` and `m` are `Integer`s, and they typecheck as `Double`s if `s` and `m` are `Double`s, because they are used in the same expressions. I think this is confusing. What do you think about explicitly annotating `([0..3] :: [Double])`? This is verbose, but not confusing.

Reply:
It feels very awkward to do `cycle cs !! floor y`. Even though it will probably work, it smells like rounding errors. I'd rather have `x` and `y` firmly `Int`s, and then cast them to `Double` wherever necessary.

Reply:
Okay, so tell me how I should do this now. Should `s` and `m` be `Double`s or `Integer`s? I think we already decided that they should be `Double`s, just making sure. From the last comments I see that `x` and `y` should be `Int`s, which makes sense. I'll change it and update the docs, and hopefully this PR will be good after that.

Reply:
* `x, y :: Int`
* `s` and `m` should have the type that everything is as simple as possible, so probably `Double`

(Sorry for going back and forth a bit)